src/sort.ts

```diff
@@ -35,10 +35,8 @@ export const DEFAULT_SORT_OP = 'mean';
 export interface EncodingSortField<F> {
   /**
    * The data [field](https://vega.github.io/vega-lite/docs/field.html) to sort by.
-   *
-   * __Default value:__ If unspecified, defaults to the field specified in the outer data reference.
    */
-  field?: F;
+  field: F;
```

Comment:
Field is not required when the "op" is count.  So this would be also incorrect. 

To make the typing perfectly correct, we need to separate count from other aggregate in sort field. However, that will cause a problem downstream in a library such as Atlair as they rely on a class called EncodingSortField.  

Thus, I think the current syntax as is seems to be the least problematic solution.